ATen/native (5/6): Use per-operator headers

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75576
* __->__ #75575
* #75574
* #75573

Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

peterbell10 committed:
ATen/native (5/6): Use per-operator headers

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and remove unneeded _meta.h ops headers on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"




[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40126696).

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Rebase on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40126696).

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Rebase on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Remove assert from NestedTensorMath.cpp as its creating too many merge conflicts on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40126696).

peterbell10 committed:
Rebase on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

peterbell10 committed:
Update on "ATen/native (5/6): Use per-operator headers"


Differential Revision: [D40126696](https://our.internmc.facebook.com/intern/diff/D40126696)

[ghstack-poisoned]

malfet commented:
This break trunk (I guess land race):
```
var/lib/jenkins/workspace/aten/src/ATen/native/TensorShape.cpp:92:10: fatal error: ATen/ops/diag_backward_native.h: No such file or directory
 #include <ATen/ops/diag_backward_native.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
